actors/stage/src/main/java/com/ea/orbit/actors/runtime/Messaging.java

```diff
@@ -93,27 +94,39 @@ public NodeAddress getNodeAddress()
         return clusterPeer.localAddress();
     }
 
-    private static class PendingResponse extends Task<Object> implements Comparable<PendingResponse>
+    /**
+     * The messageId is used only to break a tie between messages that timeout at the same ms.
+     */
+    static class PendingResponseComparator implements Comparator<PendingResponse>
     {
-        long timeoutAt;
-        public int messageId;
-
         @Override
-        public int compareTo(final PendingResponse o)
+        public int compare(final PendingResponse o1, final PendingResponse o2)
         {
-            int cmp = Long.compare(timeoutAt, o.timeoutAt);
+            int cmp = Long.compare(o1.timeoutAt, o2.timeoutAt);
             if (cmp == 0)
             {
-                return messageId - o.messageId;
+                return o1.messageId - o2.messageId;
             }
             return cmp;
         }
+    }
 
-        @Override
-        public boolean equals(final Object o)
```

Comment:
Since the equals / hashCode contract is broken just to keep FindBugs happy I extracted the compare to a Comparator.